app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuSitePermissionsCameraRobot.kt

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.ui.robots
+
+/**
+ * Implementation of Robot Pattern for the settings Site Permissions Camera sub menu.
+ */
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.CoreMatchers
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.assertIsChecked
+import org.mozilla.fenix.helpers.click
+
+class SettingsSubMenuSitePermissionsCameraRobot {

```

Comment:
So this Robot is almost equal to `SitePermiissioonsLocationRobot` and `SitePermissionsMicrophoneRobot` except for the header and one string in option 3. I think there may be a way to define one robot and parametrize the exceptions for example. There is a lot of not duplicate but even triplicate code, we should avoid that...

Reply:
ok sure. When I enquired Richard regarding creation of robots I was asked to create a separate robot for each and every sub items. So, I thought in this way I should this. May be I misunderstood his explaination

Reply:
Yes, that's the general rule, but in this case, these robots are almost identical and not very commonly used, so we will be save by adding just one.

Reply:
ok I will do that. Thanks for the check!